src/CoreFoundation/CFPropertyList.cs

```diff
@@ -45,6 +42,51 @@ public CFPropertyList (IntPtr handle)
 			this.handle = handle;
 		}
 
+		[DllImport (Constants.CoreFoundationLibrary)]
+		static extern IntPtr CFPropertyListCreateWithData (IntPtr allocator, IntPtr dataRef, CFPropertyListMutabilityOptions options, out CFPropertyListFormat format, /* CFError * */ out IntPtr error);
+
+
+		public (CFPropertyList propertyList, CFPropertyListFormat format, NSError error)
+			FromData (NSData data, CFPropertyListMutabilityOptions options = CFPropertyListMutabilityOptions.Immutable)
+		{
+			if (data == null)
+				throw new ArgumentNullException (nameof (data));
+			if (data.Handle == IntPtr.Zero)
+				throw new ObjectDisposedException (nameof (data));
+
+			CFPropertyListFormat fmt;
+			IntPtr error;
+			var ret = CFPropertyListCreateWithData (IntPtr.Zero, data.Handle, options, out fmt, out error);
+			if (ret != null)
+				return (new CFPropertyList (ret), fmt, null);
+			return (null, CFPropertyListFormat.XmlFormat1, new NSError (error));
+		}
+
+		[DllImport (Constants.CoreFoundationLibrary)]
+		extern static IntPtr CFPropertyListCreateDeepCopy(IntPtr allocator, IntPtr propertyList, CFPropertyListMutabilityOptions mutabilityOption);
+
+		public CFPropertyList DeepCopy (CFPropertyListMutabilityOptions options = CFPropertyListMutabilityOptions.MutableContainersAndLeaves)
+		{
+			return new CFPropertyList (CFPropertyListCreateDeepCopy (IntPtr.Zero, handle, options));
+		}
+
+		[DllImport (Constants.CoreFoundationLibrary)]
+		extern static /*CFDataRef*/IntPtr CFPropertyListCreateData(IntPtr allocator, IntPtr propertyList, CFPropertyListFormat format, CFPropertyListMutabilityOptions options, out IntPtr error);
+
+		public NSData AsData (CFPropertyListFormat format = CFPropertyListFormat.BinaryFormat1)
+		{
+			IntPtr error;
+			return new NSData (CFPropertyListCreateData (IntPtr.Zero, handle, format, 0, out error));
```

Comment:
I've got a question here. How is the user going to access the error if it occurrs? I think we should surface that.